go/store/prolly/tree/immutable_tree.go

```diff
@@ -0,0 +1,151 @@
+// Copyright 2022 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tree
+
+import (
+	"context"
+	"errors"
+	"io"
+
+	"github.com/dolthub/dolt/go/store/prolly/message"
+)
+
+const defaultFixedChunkLength = 4000
+
+var ErrInvalidChunkSize = errors.New("invalid chunkSize; value must be > 1")
+
+// buildImmutableTree writes the contexts of |reader| as an append-only
+// tree, returning the root node or an error if applicable. |chunkSize|
+// fixes the split size of leaf and intermediate node chunks.
+func buildImmutableTree(ctx context.Context, r io.Reader, ns NodeStore, S message.Serializer, chunkSize int) (Node, error) {
+	if chunkSize <= 1 {
```

Comment:
1 byte is invalid?

Reply:
1 byte is OK and tested, chunkSize 1 causes infinite recursion (can't be tree)